compare_template_text/normalize_template_text.py

```diff
@@ -32,67 +33,53 @@ def remove_omitable_text(self):
         which is optional and present matches and gets replaced by '' while the text that doesn't match will 
         remain as it is thus giving an unmatch. """
          
-        lists = re.findall('(?<=\<\<beginoptional\>\>).*?(?=\<\<endoptional\>\>)',
+        lists = re.findall('(?<=<\<beginoptional\>\>).*?(?=\<\<endoptional\>\>)',
                            self.template_string)
-        # print(lists)
+        print(lists)
         for x in lists:
-            if(x.startswith('*')):
+            if(x.startswith('`')):
                 x = x[1:]
 
-            if(x.endswith('*')):
+            if(x.endswith('`')):
                 x = x[:-1]
-
-            self.text_string = self.text_string.replace(x,'')
-            self.template_string = self.template_string.replace(x,'')
-            self.template_string = self.template_string.replace('<<beginoptional>>','')
-            self.template_string = self.template_string.replace('<<endoptional>>','')
+            # print(x)
+            self.template_string = self.template_string.replace(x,'(`)?('+x+')?(`)?')
```

Comment:
The optional string may include characters that would need to be escaped for a regular expression.  You will need to escaped.  You can use the `re.escape(pattern)` method.

Reply:
Actually, since you are using normalized template as a regex, you'll need to escape all of the text.  Perhaps you can call `re.escape` at the start of this method which will escape all the text the process the escaped text for optional.

Reply:
Done the changes. Please check for both the replaceable and the omitable text.